e2e/vertical-full-page-map.spec.ts

```diff
@@ -0,0 +1,96 @@
+import { test, expect } from '@playwright/test';
+
+test.describe('full page map test suite', () => {
+  test.beforeEach(async ({page}) => {
+    await page.goto('http://localhost:5042/locations_full_page_map');
+    await page.getByPlaceholder('Search for locations').click();
+  });
+
+  test('can search and get results', async ({ page }) => {
+    await page.getByPlaceholder('Search for locations').fill('virginia');
+    await page.getByPlaceholder('Search for locations').press('Enter');
+    const count = await page.locator('#js-answersVerticalResults').count();
+    expect(count).toBeGreaterThan(0);
+  });
+
+  test('clicking on a pin focuses on a result card', async ({ page }) => {
+    await page.getByPlaceholder('Search for locations').fill('virginia');
+    await page.getByPlaceholder('Search for locations').press('Enter');
+    await page.getByRole('button', { name: 'Result number 5' }).click();
+    const locator = page.locator('#js-answersVerticalResults div').nth(134);
+    await expect(locator).toHaveClass(/pinFocused/);
+  });
+
+  test('search when map moves works', async ({ page }) => {
+    const searchLogo = '#js-yext-submit';
+    await page.mouse.move(600, 300);
+    await page.mouse.down();
+    await page.mouse.move(1200, 450, {steps: 5});
+    await page.mouse.up();
+    await page.waitForSelector(searchLogo, { state: 'detached' });
+    const detachedSearchLogo = await page.$(searchLogo);
+    expect(detachedSearchLogo).toBeFalsy();
+  });
+
+  test('search this area button works', async ({ page }) => {
+    await page.getByPlaceholder('Search for locations').fill('virginia');
+    await page.getByPlaceholder('Search for locations').press('Enter');
+    const responsePromise = await page.waitForResponse(/https:\/\/prod-cdn\.us\.yextapis\.com\/v2\/accounts\/me\/search\/vertical/i);
+    await page.getByLabel('Map controls').getByText('Search When Map Moves').click();
+    const response = responsePromise;
+  });
+
+  test('default initial search works and is enabled by default', async ({ page }) => {
+    await page.getByPlaceholder('Search for locations').press('Enter');
+    const result = page.locator('#js-answersVerticalResults div').nth(0);
+    await expect(result).toBeAttached();
+  });
+
+  test('pagination works', async ({ page }) => {
+    await page.getByPlaceholder('Search for locations').fill('virginia');
+    await page.getByPlaceholder('Search for locations').press('Enter');
+    await page.getByLabel('Go to the next page of results').click();
+    const secondPage = page.locator('#js-answersVerticalResultsCount');
+    await expect(secondPage).toHaveText(/21/);
```

Comment:
I'm cautious about this check because this would break if an entity is added to the account because it would change the results count. Another idea is checking the query params of the web page to see if the offset changed. But if it's too much effort I think we can stick with this because the slapshot test account is pretty stable

Reply:
I used 21 because the offset is set to 20 for each page, so i think as long as the results size is more than 20, even if the total results count changes the check for 21 should work, but let me know if i am misunderstanding your comment!